src/providers/analytics/provider.ts

```diff
@@ -34,49 +44,64 @@ export class IndexAnalyticsProvider implements AnalyticsProvider {
     private readonly coingeckoService: CoinGeckoService,
   ) {}
 
-  async getAnalytics(address: string): Promise<IndexAnalytics> {
-    const { baseCurrency, coingeckoService, provider } = this
+  async getAnalytics(
+    address: string,
+    options: IndexAnalyticsOptions = {
+      includeTotalSupply: true,
+      include24hrVolume: true,
+    },
+  ): Promise<IndexAnalytics> {
+    const { baseCurrency, chainId, coingeckoService, provider } = this
     const marketCapProvider = new IndexMarketCapProvider(
       provider,
       coingeckoService,
     )
     const navProvider = new IndexNavProvider(provider, coingeckoService)
-    const supplyProvider = new IndexSupplyProvider(provider)
-    // For now we can save time by always assuming Ethereum
-    const chainId = 1
-    // Token data fetched statically to avoid lengthy contract interactions
-    const token = TokenData[address.toLowerCase()]
-    const totalSupply = await supplyProvider.getSupply(address)
-    const coingeckoRes = await coingeckoService.getTokenPrice({
+
+    const supplyPromise = options.includeTotalSupply
+      ? new IndexSupplyProvider(provider).getSupply(address)
+      : Promise.resolve(null)
+    const marketCapPromise = marketCapProvider.getMarketCap(address)
+    const navPricePromise = navProvider.getNav(address)
+    const coingeckoPromise = coingeckoService.getTokenPrice({
       address,
       chainId,
       baseCurrency,
       include24hrChange: true,
-      include24hrVol: true,
+      include24hrVol: options.include24hrVolume ?? false,
     })
-    const coingeckoData = coingeckoRes[address.toLowerCase()]
-    const change24hLabel = CoinGeckoUtils.get24hChangeLabel(baseCurrency)
-    const change24h =
-      coingeckoData === undefined ? 0 : coingeckoData[change24hLabel]
-    const marketPrice =
-      coingeckoData === undefined ? 0 : coingeckoData[baseCurrency]
-    const volume24hLabel = CoinGeckoUtils.get24hVolumeLabel(baseCurrency)
-    const volume24h =
-      coingeckoData === undefined ? 0 : coingeckoData[volume24hLabel]
```

Comment:
@janndriessen one note - if coingeckoData has an issue these endpoints will now return 'null' instead of 0. I think this approach makes it easier to differentiate between an error that occurs on the error and when there really is a '0', e.g. if we get back 0 as the volume after this change we'll know there truly is no volume

wdyt?

Reply:
works for me